agents/tools/wicked_git.py

```diff
@@ -106,6 +106,41 @@ async def git_am_show_current_patch(repository_path: AbsolutePath) -> str:
     return ""
 
 
+async def discover_patch_p(patch_file_path: AbsolutePath, repository_path: AbsolutePath) -> int:
+    """
+    Process the given patch file and figure out with which `-p` value the patch should be applied
+    in the given repository.
+
+    Using `git apply --stat` we parse the given patch and try to fit it into the given repository.
+    """
+    cmd = ["git", "apply", "--stat", str(patch_file_path)]
+    exit_code, stdout, stderr = await run_subprocess(cmd, cwd=repository_path)
+    if exit_code != 0:
+        # this means the patch is borked
+        raise ToolError(f"Command git-apply --stat failed: {stderr}")
+    # expat/lib/xmlparse.c                        |    8 -
+    # .github/workflows/scripts/mass-cppcheck.sh  |    1
+    # .github/workflows/data/exported-symbols.txt |    2
+    # expat/lib/expat.h                           |   15 +
+    lines = stdout.splitlines()
+    files = [line.split("|")[0].strip() for line in lines if "|" in line]
+
+    # 0 should be impossible, git-apply hates it:
+    #   "git diff header lacks filename information when removing 1 leading pathname component (line 5)"
+    # but how about /usr/bin/patch? can it handle "0"?
+    for n in range(1, 4):  # I truly hope 3 is impossible
```

Comment:
I can see 17 spec files with `-p3` in Fedora, 10 with `-p4` and 2 with `-p5` 😅

Reply:
Oh my, and 496 with `-p0`.

Reply:
OMG, get out

(thanks for checking, I'll update it)